avocado/core/runners/sysinfo.py

```diff
@@ -0,0 +1,298 @@
+
+        self.sysinfo_files = sysinfo_files
+        self.collectibles = set()
+
+    def _set_collectibles(self):
+        for cmd in self.sysinfo_files.get("commands", []):
+            self.collectibles.add(Command(cmd))
+
+        for filename in self.sysinfo_files.get("files", []):
+            self.collectibles.add(LogFile(filename))
+
+    def _get_installed_packages(self):
+        sm = software_manager.SoftwareManager()
+        installed_pkgs = sm.list_all()
+        self._installed_pkgs = installed_pkgs
+        return installed_pkgs
+
+    def collect(self):
+        """Log all collectibles at the start of the event."""
+        self._set_collectibles()
+        for log_hook in self.collectibles:
+            log_hook.run(self.queue, self.sysinfo_dir)
+
+        if self.log_packages:
+            self._log_packages(self.sysinfo_dir)
+        self.queue.put(messages.FinishedMessage.get('pass'))
+
+    def _log_packages(self, path):
+        installed_path = os.path.join(path, "installed_packages")
+        installed_packages = "\n".join(self._get_installed_packages()) + "\n"
+        self.queue.put(messages.FileMessage.get(installed_packages,
+                                                installed_path))
+
+
+class PostSysInfo(PreSysInfo):
+    """
+    Log different system properties after end event.
+
+    An event may be a job, a test, or any other event with a
+    beginning and end.
+    """
+
+    sysinfo_dir = os.path.join('sysinfo', 'post')
+
+    def __init__(self, config, sysinfo_files, queue, test_fail=False):
+        """
+        :param test_fail: flag for fail tests. Default False
+        :type test_fail: bool
+        """
+        self.test_fail = test_fail
+        super().__init__(config, sysinfo_files, queue)
+
+    def _set_collectibles(self):
+        super()._set_collectibles()
+        if self.test_fail:
+            for fail_cmd in self.sysinfo_files.get("fail_commands", []):
+                self.collectibles.add(Command(fail_cmd))
+
+            for fail_filename in self.sysinfo_files.get("fail_files", []):
+                self.collectibles.add(LogFile(fail_filename))
+
+
+class SysinfoRunner(nrunner.BaseRunner):
+    """
+    Runner for gathering sysinfo
+
+    Runnable attributes usage:
+
+     * uri: Specification of sysinfo. Possible values pre/post
+
+     * kwargs: sysinfo_files dictionary with commands/tasks which should be
+             performed during the sysinfo collection.
+    """
+
+    def run(self):
+        yield self.prepare_status('started')
+        sysinfo_files = self.runnable.kwargs.get('sysinfo_files', {})
+        test_fail = self.runnable.kwargs.get('test_fail', False)
+        try:
+            queue = multiprocessing.SimpleQueue()
+            if self.runnable.uri == 'pre':
+                sysinfo = PreSysInfo(self.runnable.config,
+                                     sysinfo_files,
+                                     queue)
+            else:
```

Comment:
This doesn't match the description on the URI, an "else if" would be needed here.